Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -35,1657 +35,1664 @@
+
+                        carve::geom::vector<3>  translate(carve::geom::VECTOR(0.0, 0.0, 0.0));
+                        carve::geom::vector<3>  local_x(carve::geom::VECTOR(1.0, 0.0, 0.0));
+                        carve::geom::vector<3>  local_y(carve::geom::VECTOR(0.0, 1.0, 0.0));
+                        carve::geom::vector<3>  local_z(carve::geom::VECTOR(0.0, 0.0, 1.0));
+                        carve::geom::vector<3>  ref_direction(carve::geom::VECTOR(1.0, 0.0, 0.0));
+
+                        // interpret Location
+                        convertIfcCartesianPoint(axis2placement3d->Location.lock(), translate);
+
+                        // interpret RefDirection [OPTIONAL]
+                        if(axis2placement3d->RefDirection) {
+                            convertIfcDirection(axis2placement3d->RefDirection.get().lock(), ref_direction);
+                        }
+
+                        // interpret Axis [OPTIONAL]
+                        if(axis2placement3d->Axis) {
+                            convertIfcDirection(axis2placement3d->Axis.get().lock(), local_z);
+                        }
+
+                        // calculate
+                        local_x = ref_direction;
+                        // ref_direction can be just in the x-z-plane, not perpendicular to y and z.
+                        //  --> so re-compute local x					
+                        local_y = carve::geom::cross(local_z, local_x);
+                        local_x = carve::geom::cross(local_y, local_z);
+
+                        // normalize the direction vectors
+                        local_x.normalize();
+                        local_y.normalize();
+                        local_z.normalize();
+
+                        // produce a quaternion
+                        matrix = carve::math::Matrix(
+                            local_x.x, local_y.x, local_z.x, translate.x,
+                            local_x.y, local_y.y, local_z.y, translate.y,
+                            local_x.z, local_y.z, local_z.z, translate.z,
+                            0, 0, 0, 1);
+                    }
+
+                    // The implementation is in file PlacementConverterImpl.h
+                    std::shared_ptr<typename IfcEntityTypesT::IfcCurve> GetCurveOfPlacement(
+                        const std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement>& linearPlacement);
+
+
+
+                    /*! \brief Retrieves the respective placement type from the \c IfcLocalPlacement and converts it.
+
+                    \param[in]	    local_placement		\c IfcLocalPlacement entity to be interpreted.
+                    \param[in,out]  matrix			Calculated transformation matrix.
+
+                    Retrieves the respective placement type from the \c IfcLocalPlacement and converts it.
+                    The result is stored in matrix. Uses the switch statement to get the correct type.
+                    */

```

Comment:
Needs to be adapted to the function.

Reply:
What has to be adapted? The comment describes what the function does.